src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private Object[] elements;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    private void resize() {
+        capacity = (int) (capacity * 1.5);
+        elements = Arrays.copyOf(elements, capacity);
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            resize();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("The index " + index + " is unavailable");
```

Comment:
The index validation logic is repeated here and in other methods like `get`, `set`, and `remove`. According to the checklist, 'If the logic of your code repeats - move it to a separate private method.' Consider creating a private method to handle index validation .